bokehjs/src/coffee/document.coffee

```diff
@@ -12,6 +12,25 @@ import {isEqual} from "./core/util/eq"
 import {isArray, isObject} from "./core/util/types"
 import {ColumnDataSource} from "./models/sources/column_data_source"
 
+class EventManager
+    # Dispatches events to the subscribed models
+
+  constructor: (@document) ->
+    @session = null
+    @subscribed_models = new Set()
+
+  send_event: (event) ->
+    # Send message to Python via session
+    @session.send_event(event)
+
+  trigger: (event) ->
+    for model_id in @subscribed_models.values
+        if event.model_id != null and event.model_id != model_id
+            continue
+        model = @document._all_models[model_id]
+        if model?
+            model._process_event(event)
```

Comment:
Now addressed - file is now converted to typescript with the correct indentation.